doc/source/serve/deployment.rst

```diff
@@ -330,32 +330,88 @@ In this example pictured above, we have a single-node cluster with a deployment
+
+In development, logs are streamed to the driver Ray program that deployed the deployments, so it's most convenient to keep the driver running for debugging.
+For example, let's run a basic Serve application and view the logs that are emitted.
+You can run this in an interactive shell like IPython to follow along.
+
+First we call ``serve.start()``:
+
+.. code-block:: python
+
+   from ray import serve
+
+   serve.start()
+
+This produces a few INFO-level log messages about startup from the Serve controller.
+
+.. code-block:: bash
+
+   2022-04-02 09:10:49,906 INFO services.py:1460 -- View the Ray dashboard at http://127.0.0.1:8265
+   (ServeController pid=67312) INFO 2022-04-02 09:10:51,386 controller 67312 checkpoint_path.py:17 - Using RayInternalKVStore for controller checkpoint and recovery.
+   (ServeController pid=67312) INFO 2022-04-02 09:10:51,492 controller 67312 http_state.py:108 - Starting HTTP proxy with name 'SERVE_CONTROLLER_ACTOR:xlehoa:SERVE_PROXY_ACTOR-node:127.0.0.1-0' on node 'node:127.0.0.1-0' listening on '127.0.0.1:8000'
+
+Next, let's create a simple deployment that logs a custom log message when it's queried:
+
+.. code-block:: python
+
+   import logging
+
+   logger = logging.getLogger("ray.serve")
+
+   @serve.deployment(route_prefix="/")
+   class SayHello:
+       def __call__(self, *args):
+           logger.info("Hello world!")
+           return "hi"
+   
+   SayHello.deploy()
+   handle = SayHello.get_handle()
+   handle.remote()
+
+Running this code block, we first get some log messages from the controller saying that a new replica of the deployment is being created:
+
+.. code-block:: bash
 
-Querying the above deployment will produce a log line like the following:
+   (ServeController pid=67312) INFO 2022-04-02 09:16:13,323 controller 67312 deployment_state.py:1198 - Adding 1 replicas to deployment 'SayHello'.
+
+Then when we query the deployment, we get both a default access log as well as our custom ``"Hello world!"`` message.
+Note that these log lines are tagged with the deployment name followed by a unique identifier for the specific replica.
+These can be parsed by a logging stack such as ELK or Loki to enable searching logs by deployment and replica.
+
+Querying the deployment over HTTP produces a similar access log message from the HTTP proxy:
 
 .. code-block:: bash
 
-  (pid=42161) 2021-02-26 11:05:21,709     INFO snippet_logger.py:13 -- Some info! component=serve deployment=f replica=f#jZlnUI
+   curl -X GET http://localhost:8000/
+   (HTTPProxyActor pid=67315) INFO 2022-04-02 09:20:08,976 http_proxy 127.0.0.1 http_proxy.py:310 - GET / 200 2.6ms
+   (SayHello pid=67352) INFO 2022-04-02 09:20:08,975 SayHello SayHello#LBINMh <ipython-input-4-1e8854e5c9ba>:8 - Hello world!
+   (SayHello pid=67352) INFO 2022-04-02 09:20:08,975 SayHello SayHello#LBINMh replica.py:466 - HANDLE __call__ OK 0.3ms
+
```

Comment:
what happen if users don't use the ray.serve logger? do they go somewhere? worth mentioning as this must be the most common error.

Reply:
let me add!